model-engine/model_engine_server/common/dtos/llms/vllm.py

```diff
@@ -0,0 +1,146 @@
+    )
+    use_beam_search: Optional[bool] = Field(
+        None,
+        description="""Whether to use beam search for sampling.""",
+    )
+    length_penalty: Optional[float] = Field(
+        default=None,
+        description="""Float that penalizes sequences based on their length.
+            Used in beam search.""",
+    )
+    repetition_penalty: Optional[float] = Field(
+        default=None,
+        description="""Float that penalizes new tokens based on whether
+            they appear in the prompt and the generated text so far. Values > 1
+            encourage the model to use new tokens, while values < 1 encourage
+            the model to repeat tokens.""",
+    )
+    early_stopping: Optional[bool] = Field(
+        None,
+        description="""Controls the stopping condition for beam search. It
+            accepts the following values: `True`, where the generation stops as
+            soon as there are `best_of` complete candidates; `False`, where an
+            heuristic is applied and the generation stops when is it very
+            unlikely to find better candidates; `"never"`, where the beam search
+            procedure only stops when there cannot be better candidates
+            (canonical beam search algorithm).""",
+    )
+    stop_token_ids: Optional[List[int]] = Field(
+        default_factory=list,
+        description="""List of tokens that stop the generation when they are
+            generated. The returned output will contain the stop tokens unless
+            the stop tokens are special tokens.""",
+    )
+    include_stop_str_in_output: Annotated[
+        Optional[bool],
+        Field(
+            None,
+            description="""Whether to include the stop strings in
+            output text. Defaults to False.""",
+        ),
+    ]
+    ignore_eos: Optional[bool] = Field(
+        None,
+        description="""Whether to ignore the EOS token and continue generating
+            tokens after the EOS token is generated.""",
+    )
+    min_tokens: Optional[int] = Field(
+        None,
+        description="""Minimum number of tokens to generate per output sequence
+            before EOS or stop_token_ids can be generated""",
+    )
+
+    skip_special_tokens: Optional[bool] = Field(
+        True,
+        description="Whether to skip special tokens in the output. Only supported in vllm.",
+    )
+
+    spaces_between_special_tokens: Optional[bool] = Field(
+        True,
+        description="Whether to add spaces between special tokens in the output. Only supported in vllm.",
+    )
+
+
+class VLLMChatCompletionAdditionalParams(VLLMSamplingParams):
+    chat_template: Optional[str] = Field(
+        default=None,
+        description=(
+            "A Jinja template to use for this conversion. "
+            "As of transformers v4.44, default chat template is no longer "
+            "allowed, so you must provide a chat template if the tokenizer "
```

Comment:
should this be "if the model's tokenizer does not define one and no override template is given"?